### PR TITLE
Animated Themes. Subthemes. Custom default_theme. Overhaul asset paths. Fix a ton of asset resolution bugs.

### PR DESCRIPTION
**Pull request by Crystalwarrior**
This overhauls the asset/path resolution of AO2, making it much nicer and make a lot more sense. Most importantly, it drastically reduces code duplication, causing less inconsistencies and fixes a metric crapton of bugs related to that.

`NEEDS TESTING!`
 - [x] Implements default_theme courtroom_config.ini option 😲 
 - [x] COMPLETELY overhauls the asset/path resolution, making it feel like an actually decent program now
 - [x] fixes a metric crapton of bugs, see commit history for specifics
 - [x] fixes a very old bug that caused the viewport to freeze because it never received the done(); signal it needs from objections, characters, etc.
 - [x] Implements subthemes, and an "ST" packet. "ST" "subtheme" "1" where subtheme is a string you want your client to have their sub-theme set to, and "1" stands for if you want to reload their theme or not.
 - [x] Subtheme setting - "server", "default", and the available subthemes in any given theme. "Server" prioritizes the subtheme we were requested by the server. "Default" means no subtheme will be initialized.
 - [x]  Adds optional animated themes which allows you to have animated buttons, images, etc.

Screenshot

Closes https://github.com/AttorneyOnline/AO2-Client/issues/427

New asset resolution functions:
```get_asset_paths - Return an untested list of universal paths from the provided args
get_asset_path - Loop through the list of provided asset paths and return the first valid file
get_image_path - Loop through the list of provided asset paths, apply get_image_suffix and return the first valid image file
get_sfx_path - Loop through the list of provided asset paths, apply get_sfx_suffix and return the first valid sound file
get_asset - return an asset (must contain file extension) from the get_asset_path() applied on the get_asset_paths()
get_image - return an image with get_image_suffix() applied on the get_image_path() for the get_asset_paths()
get_sfx - return a sfx from provided args with the uniquely constructed asset path list for sounds
get_config_value - obtains a value from the config that matches identifier, iterating through the list of valid paths```

### CODE REVIEW

**Comment by Crystalwarrior** [comment]
I will implement the subthemes in a separate PR. This PR is done but requires review and testing, especially since it overhauls the asset resolution system and introduces `default_theme` - it also fixes a very old bug that caused the viewport to freeze because it never received the done(); signal it needs from objections, characters, etc.

> Subtheme explanation:
Completely fuckin forgot to add this to 2.9, but I've been talking about this feature since the update's inception.
The intended mechanics of the subthemes system is as such:
 - [ ] Users can select a subtheme in the Settings screen. It stacks whatever is inside of the subtheme folder on top of the current theme.
 - [ ] The server can send a packet demanding a certain subtheme. If the subtheme exists, it will be applied on the user if they have not overridden it in the settings.
 - [ ] In essence, a subtheme is exactly the same as normal themes, but the subtheme's "default" variant is the current theme it's nested in rather than themes/default/

**Review by in1tiate — APPROVED** [review]
poggers

File: src/path_functions.cpp [hunk 1]
```diff
@@ -112,18 +102,132 @@ QString AOApplication::get_evidence_path(QString p_file)
...
+      pathlist += get_theme_path("misc/" + p_misc + "/" + p_sfx, current_theme + "/" + get_subtheme()); // Subtheme misc path
+    if (!p_misc.isEmpty())
+      pathlist += get_theme_path("misc/" + p_misc + "/" + p_sfx, current_theme); // Theme misc path
+    if (!get_subtheme().isEmpty())
+      pathlist += get_theme_path(p_sfx, current_theme + "/" + get_subtheme()); // Subtheme path
+    if (!p_misc.isEmpty())
+      pathlist += get_misc_path(p_misc, p_sfx); // Base misc path
+    pathlist += get_theme_path(p_sfx, current_theme); // Theme path
+    pathlist += get_theme_path(p_sfx, default_theme); // Default theme path
+    pathlist += get_sounds_path(p_sfx); // Sounds folder path
```
Comment (oldmud0) [inline src/path_functions.cpp:216]: excuse me but is this not a copypaste of `AOApplication::get_asset_paths`

Comment (Crystalwarrior) [reply src/path_functions.cpp:216]: uhhhhhhhh

File: include/aoimage.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 #include <QDebug>
 #include <QLabel>
+#include <QMovie>
```
Comment (in1tiate) [inline include/aoimage.h:10]: cursed

**Review by in1tiate — APPROVED** [review]
Looks good, but needs documentation.

**Comment by Crystalwarrior** [comment]
> 
> 
> Looks good, but needs documentation.

I suck at documentation :((((

**Pull request merged**